prow/kube/config.go

```diff
@@ -98,10 +101,44 @@ func LoadClusterConfigs(kubeconfig, projectedTokenFile string) (map[string]rest.
 		logrus.WithField("tokenfile", projectedTokenFile).Info("Using projected token file")
 	}
 
-	kubeCfgs, currentContext, err := kubeConfigs(kubeconfig)
-	if err != nil {
-		return nil, fmt.Errorf("kubecfg: %v", err)
+	var candidates []string
+	if kubeconfig != "" {
+		candidates = append(candidates, kubeconfig)
+	}
+	if kubeconfigDir != "" {
+		files, err := ioutil.ReadDir(kubeconfigDir)
+		if err != nil {
+			return nil, fmt.Errorf("kubecfg dir: %v", err)
+		}
+		for _, file := range files {
+			filename := file.Name()
+			if kubeconfigSuffix != "" && !strings.HasSuffix(filename, kubeconfigSuffix) {
+				continue
+			}
+			candidates = append(candidates, filepath.Join(kubeconfigDir, file.Name()))
+		}
+	}
+
+	if len(candidates) == 0 {
+		// loading from ${KUBECONFIG}
+		candidates = append(candidates, "")
```

Comment:
This is (existing) bad factoring. We should have a function that knows how to load data from files, and a separate function that knows how to load data from environment variables. We should not pass around empty strings to indicate what we want

Reply:
Done.